app/models/section.rb

```diff
@@ -61,6 +67,13 @@ def generate_home_node
       end
     end
   end
+
+
+  def generate_news_node
+    unless news_node.present?
+      # TODO: Create NewsNode
```

Comment:
Yeah a more generic version might be good. I'll create the generic indispensable version and leave refactoring generate_home_nodes as a future task.